### PR TITLE
19024 - Implemented the Involuntary Dissolution Screen

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:*
https://github.com/bcgov/entity/issues/19024

### Designs: https://www.figma.com/file/LcP4NgVYKVxEk8V7fXWIuc/Involuntary-Dissolution-%26-Delay-of-Dissolution?type=design&node-id=390-3159&mode=design

*Description of changes:*
- Implemented the Involuntary Dissolution Screen
- Implemented all phases (pause, resume, edit)
- Implemented the badge
- Implemented validation
- Lots of misc. stuff

staff dissolution batch paused

staff dissolution batch edit

### _NOTE: Unit tests will be done in https://github.com/bcgov/entity/issues/21154_

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/assets/scss/theme.scss
Comment (JazzarKarim) [inline auth-web/src/assets/scss/theme.scss:78]: This'll be the background color of the new badge:
paused bg color

File: auth-web/src/components/CardHeader.vue
Comment (JazzarKarim) [inline auth-web/src/components/CardHeader.vue:1]: The Card header:
card header

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2805-ip6uoedq.web.app

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (JazzarKarim) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:49]: The Tuesday and hour here will probably change later.

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (JazzarKarim) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:140]: We usually use the `@Getter` and `@Action` in our entities codebases but in here, I'm just following the coding practice of relationships to stay uniform across all files 😃

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (JazzarKarim) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:156]: You may notice that I'm using the onHold a lot. This is because, from the configurations table (got this from Argus), we'll be getting something like this:
```json
{
  "configurations": [
    {
      "name": "NUM_DISSOLUTIONS_ALLOWED",
      "value": "100"
    },
    {
      "name": "DISSOLUTIONS_ON_HOLD",
      "value": false
    },
   ...
  ]
}
```

So, I'm using the on hold term.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2805-ip6uoedq.web.app

File: auth-web/package.json
Comment (JazzarKarim) [inline auth-web/package.json:23]: @ochiu Ody, I noticed this change was reverted at one point: https://github.com/bcgov/sbc-auth/pull/2716

What kind of issues were you running into exactly? I'm thinking I need to change this because I need the `mdi-delete-clock` icon in the CardHeader (line 45 in the `InvoluntaryDissolution.vue`). Please let me know!

Comment (ochiu) [reply auth-web/package.json:23]: @JazzarKarim I believe there were icons we are currently using that have been removed in newer versions, so I had to revert the changes as it affected other screens.

Comment (ochiu) [reply auth-web/package.json:23]: Example:

Your PR:
Screenshot 2024-05-07 at 11 23 50 AM

Dev:
Screenshot 2024-05-07 at 11 24 20 AM

Comment (JazzarKarim) [reply auth-web/package.json:23]: I see, OK. Got it. Thank you so much Ody. 

I will talk to our UX designer and will see if we can use something else in this case. Appreciate it!

Comment (severinbeauvais) [reply auth-web/package.json:23]: You can also pull in icons separately if the font doesn't have them.

File: auth-web/src/assets/scss/theme.scss
Comment (severinbeauvais) [reply auth-web/src/assets/scss/theme.scss:78]: I suggest a more generic name so it can be reused.

Cause this would look weird: `<icon name="pumpkin" color="paused-warning" />" 😁

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:3]: This can be your top-level block. The div is useful if there are multiple children, but you only have 1.

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:195]: The seems a little suss because it's used in so many ways. Passing in the new isEdit value might be easier to understand and test. Did you test all possible flows?

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:185]: How are you keeping track of this TODO?

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:206]: Did you mean to have the same 3 error messages?

The second check is for an even/odd value, right? The message doesn't seem to convey the rule.

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:140]: I was going to comment on this, not because I think mapActions() is bad, but because you also need to declare the method signatures on lines 158-161. DRY if possible.

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:242]: Yes but for what purpose?

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:250]: 👍

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:287]: Somewhere, we declared a base class `font-size-14` to do this in the template instead of needed CSS here. YMMV

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/DissolutionSchedule.vue:10]: What does this look like on a narrow screen? I don't we don't try hard to be responsive but you might be able to copy some props from other UIs so that these columns stack nicely, with nice whitespace.

File: auth-web/src/stores/staff.ts
Comment (severinbeauvais) [inline auth-web/src/stores/staff.ts:28]: On the plus side, `involuntaryDissolutionBatch.batchSize` won't throw an error. On the down side, you can't check if `involuntaryDissolutionBatch` is null. Depends how you use it, of course, but it's something to watch out for.

An alternative is to initialize the properties here as well.
```
involuntaryDissolutionBatch: {
  batchSize: null // or NaN
  onHold: null // or False
}
```

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:185]: This comment is in a lot of places. Do you have a follow-up ticket with todos for all places that need to be updated?

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:195]: Yes. I've tested thoroughly all the flows.

I thought the comment basically explains it. It'll be triggered when the Edit, Cancel, or Save (successful) buttons are clicked to show the extra fields.

On that note, I'll update the comment to include the Save button as well. Thanks for bringing it up!

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:185]: The TODOs will all be worked on in this ticket: https://github.com/bcgov/entity/issues/20929

I'm going to grab it and work on it eventually. So, in my head, I have everything in place 😆

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:206]: The Designs didn't show different error messages but showed them as errors. The second one is to check if the value has a decimal point.

Now that I'm thinking about, it's probably a good idea to bring it up to Andy to see if we're going to show different messages.

Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:206]: Oh yeah. Checking for even/odd would be `% 2`.

You could also use `Number.isInteger()`. I'm OK with either.

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:10]: I tried to test and make it look good as much as possible by playing with the v-cols throughout it all. This is what I have as of right now on a narrow screen:
narrow screen

I think it looks good. Thoughts?

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:242]: "Set colour of ..." ?

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:10]: LGTM

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue:242]: Yep, I have a PR up in a sec.

**Review by severinbeauvais — APPROVED** [review]
I don't want to slow you down. Looks decent and it seems you've thought about my comments so I'll leave it to you to update if/as needed.

File: auth-web/src/stores/staff.ts
Comment (JazzarKarim) [reply auth-web/src/stores/staff.ts:28]: OK, will keep this in mind Sev.

I really like your suggested change. It depends on what I'll be doing here next ticket once I tie everything together. Are you OK with me keeping it as is for now?

File: auth-web/src/components/auth/staff/DissolutionSchedule.vue
Comment (JazzarKarim) [reply auth-web/src/components/auth/staff/DissolutionSchedule.vue]: Awesome catch. I updated everything this across all the new files. Thanks!

File: auth-web/src/stores/staff.ts
Comment (severinbeauvais) [reply auth-web/src/stores/staff.ts:28]: You bet. Resolving.

File: auth-web/package.json
Comment (JazzarKarim) [reply auth-web/package.json]: Reverted this change to not break other stuff as Ody mentioned.

**Comment by JazzarKarim** [comment]
Got the OK to merge from Travis. Merging now.

**Pull request merged**
